src/schema/devcards.ts

```diff
@@ -0,0 +1,185 @@
+import { IResolvers } from '@graphql-tools/utils';
+import qs from 'qs';
+
+import { Context } from '../Context';
+import { traceResolverObject } from './trace';
+import { DevCardTheme, DevCard } from '../entity';
+import { NotFoundError } from '../errors';
+import { DevCardData, getDevCardData } from '../common/devcard';
+
+export interface GQLDevCard {
+  id: string;
+  userId: string;
+  createdAt: Date;
+  theme: DevCardTheme;
+  isProfileCover: boolean;
+  showBorder: boolean;
+}
+
+export const typeDefs = /* GraphQL */ `
+  """
+  Defines available themes for dev cards
+  """
+  enum DevCardTheme {
+    DEFAULT
+    IRON
+    BRONZE
+    SILVER
+    GOLD
+    PLATINUM
+    DIAMOND
+    LEGENDARY
+  }
+
+  """
+  Determines which devcard image to generate - default, wide or the twitter/X one
+  """
+  enum DevCardType {
+    DEFAULT
+    WIDE
+    X
+  }
+
+  """
+  Dev card
+  """
+  type DevCardData {
+    """
+    ID of the dev card
+    """
+    id: String!
+
+    """
+    ID of the user that dev card belongs to
+    """
+    userId: String!
+
+    """
+    Timestamp the dev card was created
+    """
+    createdAt: DateTime!
+
+    """
+    Theme of the dev card
+    """
+    theme: String!
+
+    """
+    Whether the dev card cover is from the profile, default false
+    """
+    isProfileCover: Boolean!
+
+    """
+    Whether the dev card has a border, default true
+    """
+    showBorder: Boolean!
+
+    """
+    User reputation
+    """
+    reputation: Int!
+
+    """
+    Number of articles read
+    """
+    articlesRead: Int!
+
+    """
+    Tags
+    """
+    tags: [String!]!
+
+    """
+    Sources logo image URLs
+    """
+    sourcesLogos: [String!]!
+  }
+
+  """
+  The type returned on generateDevCard mutation
+  """
+  type DevCardImage {
+    imageUrl: String!
+  }
+
+  extend type Query {
+    """
+    Get devcard by id
+    """
+    devCard(
+      """
+      Id of the requested devcard
+      """
+      id: ID
+    ): DevCardData!
+  }
+
+  extend type Mutation {
+    """
+    Generates or updates the user's Dev Card preferences
+    """
+    generateDevCard(
+      theme: DevCardTheme
+      type: DevCardType
+      isProfileCover: Boolean
+      showBorder: Boolean
+    ): DevCardImage @auth
+  }
+`;
+
+interface GenerateDevCardInput {
+  theme: DevCard['theme'];
```

Comment:
I started with just the theme and then added more parameters. Will switch it to `extends Pick<DevCard ...` in a followup PR.

As for the enum type - this string union type is only used to parse the graphql query, where we get the data as an uppercase string and then pass it on to the URL.

The URL is [parsed in the og repo and there we use a proper enum type](https://github.com/dailydotdev/og/pull/2/files#diff-f2fdc3d96f51f3df3808d3d5d1ce10301abe8c715922645f1087c2af50aaec17R15-R19). I don't know an easy way to share the same definition in both repositories, otherwise I would.